File: filesystem/src/main/scala/net/kemitix/thorp/filesystem/FileData.scala
package net.kemitix.thorp.filesystem

import net.kemitix.thorp.domain.{Hashes, LastModified}

case class FileData(
    hashes: Hashes,
    lastModified: LastModified
) {
  def +(other: FileData): FileData = {
    FileData(
      hashes = this.hashes ++ other.hashes,
      lastModified = lastModified // discards other.lastModified
    )
  }
}

object FileData {
  def create(hashes: Hashes, lastModified: LastModified): FileData = FileData(
    hashes = hashes,
    lastModified = lastModified
  )
}

File: filesystem/src/main/java/net/kemitix/thorp/filesystem/FileData.java
package net.kemitix.thorp.filesystem;

import lombok.AccessLevel;
import lombok.RequiredArgsConstructor;
import net.kemitix.thorp.domain.Hashes;
import net.kemitix.thorp.domain.LastModified;

@RequiredArgsConstructor(access = AccessLevel.PRIVATE)
public class FileData {
    public final Hashes hashes;
    public final LastModified lastModified;
    public static FileData create(Hashes hashes, LastModified lastModified) {
        return new FileData(hashes, lastModified);
    }
    public FileData join(FileData other) {
        return FileData.create(
                hashes.merge(other.hashes),
                lastModified // discards other.lastModified
        );
    }
}
